src/test/java/com/es/phoneshop/service/impl/CartServiceImplTest.java

```diff
@@ -0,0 +1,103 @@
+package com.es.phoneshop.service.impl;
+
+import com.es.phoneshop.dao.product.ArrayListProductDao;
+import com.es.phoneshop.dao.product.ProductDao;
+import com.es.phoneshop.exception.OutOfStockException;
+import com.es.phoneshop.model.cart.Cart;
+import com.es.phoneshop.model.cart.CartItem;
+import com.es.phoneshop.model.product.SampleProduct;
+import com.es.phoneshop.service.CartService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.servlet.http.HttpSession;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertSame;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CartServiceImplTest {
+
+    final String IPHONE_CODE = "iphone";
+
+    @Mock
+    private HttpSession session;
+
+    private Cart cart;
+    private CartService cartService;
+    private ProductDao productDao;
+
+    @Before
+    public void setup() {
+        productDao = ArrayListProductDao.getInstance();
+        SampleProduct.createSampleProductsArrayList(productDao);
+        cartService = CartServiceImpl.getInstance();
+        cart = cartService.getCart(session);
+
+    }
+
+
+    @Test
+    public void GivenTwoSingletonsCartService_WhenAssertSame_ThenReturnTrue() {
+        CartService cartServiceFirst = CartServiceImpl.getInstance();
+        CartService cartServiceSecond = CartServiceImpl.getInstance();
+
+        assertSame(cartServiceFirst, cartServiceFirst);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void GivenNullCart_WhenAddInNullCart_ThenThrowNullPointerException() throws OutOfStockException {
+        Cart cart = null;
+
+        cartService.add(cart, IPHONE_CODE, 1);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void GivenQuantityBelowZero_WhenAddToCart_ThanThrowIllegalArgumentException() throws OutOfStockException {
+        final int QUANTITY = -2;
```

Comment:
You should use constant naming for constants only. Move it to class state block and define as constant or leave it here and rename with camel case.